### ISSUE TITLE
Compound primary key is not generated when a unique field has the same name

### ISSUE THREAD

**Issue by RobertCraigie** [prisma/prisma#10456]
### Bug description

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource client {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model User {
  name         String
  surname      String
  name_surname String @unique

  @@id([name, surname])
}
```

Given the above schema, Prisma generates the following type:

```ts
export type UserWhereUniqueInput = {
	name_surname?: string
}
```

If the `@@id` is given a name, e.g. 
```prisma
model User {
  name         String
  surname      String
  name_surname String @unique

  @@id([name, surname], name: "primary_key")
}
```

Then the correct type is generated:
```ts
export type UserWhereUniqueInput = {
  name_surname?: string
  primary_key?: UserPrimary_keyCompoundUniqueInput
}
```

It should be noted that if the `@@id` is explicitly given the implicit name then a validation error is shown:
```prisma
model User {
  name         String
  surname      String
  name_surname String @unique

  @@id([name, surname], name: "name_surname")
}
```
```
Error: Schema parsing
error: Error validating model "User": The custom name `name_surname` specified for the `@@id` attribute is already used as a name for a field. Please choose a different name.
  -->  schema.prisma:10
   | 
 9 | 
10 | model User {
11 |   name         String
12 |   surname      String
13 |   name_surname String @unique
14 | 
15 |   @@id([name, surname], name: "name_surname")
16 | }
   | 

Validation Error Count: 1
```

### How to reproduce

```prisma
model User {
  name         String
  surname      String
  name_surname String @unique

  @@id([name, surname])
}
```

Trying to query on the above model will fail at runtime but will pass type checks:

```ts
import { PrismaClient } from "@prisma/client";

const prisma = new PrismaClient({ log: ["query"] });

async function main() {
  const found = await prisma.user.findUnique({
    where: {
      name_surname: "foo",
    },
  });
  console.log(found);
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });
```

```
/prisma-js-starter/node_modules/@prisma/client/runtime/index.js:34745
    const error2 = new PrismaClientValidationError(renderErrorStr(validationCallsite));
                   ^
Error: 
Invalid `prisma.user.findUnique()` invocation in
/prisma-js-starter/script.ts:6:35

  3 const prisma = new PrismaClient({ log: ["query"] });
  4 
  5 async function main() {
→ 6   const found = await prisma.user.findUnique({
        where: {
          name_surname: 'foo'
                        ~~~~~
        }
      })

Argument name_surname: Got invalid value 'foo' on prisma.findUniqueUser. Provided String, expected UserNameSurnameCompoundUniqueInput:
type UserNameSurnameCompoundUniqueInput {
  name: String
  surname: String
}
```

### Expected behavior

A validation error should be shown, e.g.

```
Error: Schema parsing
error: Error validating model "User": The name `name_surname` for the `@@id` attribute is already used as a name for a field. Please choose a different name.
  -->  schema.prisma:10
   | 
 9 | 
10 | model User {
11 |   name         String
12 |   surname      String
13 |   name_surname String @unique
14 | 
15 |   @@id([name, surname])
16 | }
   | 

Validation Error Count: 1
```

### Prisma information

Schema given above.

### Environment & setup

- OS:  Mac OS
- Database:  PostgreSQL
- Node.js version:  v17.1.0

### Prisma Version

```
prisma                  : 3.5.0
@prisma/client          : 3.5.0
Current platform        : darwin
Query Engine (Node-API) : libquery-engine 78a5df6def6943431f4c022e1428dbc3e833cf8e (at node_modules/@prisma/engines/libquery_engine-darwin.dylib.node)
Migration Engine        : migration-engine-cli 78a5df6def6943431f4c022e1428dbc3e833cf8e (at node_modules/@prisma/engines/migration-engine-darwin)
Introspection Engine    : introspection-core 78a5df6def6943431f4c022e1428dbc3e833cf8e (at node_modules/@prisma/engines/introspection-engine-darwin)
Format Binary           : prisma-fmt 78a5df6def6943431f4c022e1428dbc3e833cf8e (at node_modules/@prisma/engines/prisma-fmt-darwin)
Default Engines Hash    : 78a5df6def6943431f4c022e1428dbc3e833cf8e
Studio                  : 0.439.0
```

**Comment by pantharshit00** [comment]
Thanks for reporting this, I can reproduce this. We should fix the collision here instead of throwing a validation error.

**Comment by dpetrick** [comment]
Thanks for the report, a fix will ship with 3.14.
